src/main/java/com/example/demo/service/NotificationService.java

```diff
@@ -0,0 +1,25 @@
+package com.example.demo.service;
+
+import com.example.demo.enums.AccountType;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.stereotype.Service;
+
+@Service
+public class NotificationService {
+
+    private final KafkaTemplate<String, String> kafkaTemplate;
+
+    @Autowired
+    public NotificationService(KafkaTemplate<String, String> kafkaTemplate) {
+        this.kafkaTemplate = kafkaTemplate;
+    }
+
+    public void sendWelcomeNotification(String identifier, AccountType accountType) {
+        String topic = "userTopic";
+        String msg = (accountType == AccountType.EMAIL)
+                ? "Welcome! You've registered with your email: " + identifier
+                : "Welcome! You've registered with your phone number: " + identifier;
+        kafkaTemplate.send(topic, msg);
```

Comment:
burada sadece mesaj ve AccountType.EMAIL gitmeli. sms veya mail çalışacağına da listener karar vermeli. Listener'ı ayrı bir servise koymanı istiyorum. Servisten kastım class değil ayrı bir proje.  burada kontrol ettiğin gibi bir if else değil open closed prensibine uygun bir kod bekliyorum. Yeni bir mesaj tipi eklendiğinde bu kodu değiştirmiyor olman gerekirdi fakat bu hali ile kodu her zaman manipüle etmek zorundasın